src/tests/daos_perf.c

```diff
@@ -291,7 +291,8 @@ daos_update_or_fetch(daos_handle_t oh, enum ts_op_type op_type,
 	int	rc;
 	uint64_t start = 0;
 
-	TS_TIME_START(duration, start);
+	if (ts_ctx.tsc_cred_nr < 0)
```

Comment:
It looks to me the tsc_cred_nr is always > 0, see credits_init(). Seems the proper way to check sync io is to check if tsc_cred_avail < 0 or tsc_eqh is DAOS_HDL_INVAL.

Reply:
you are right, let me refresh it, thanks.